auto_submit/lib/requests/refresh_cirrus_status.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
Instead of importing this, we should update the GitHub graphql code to query the check suites directly

See https://github.com/flutter/cocoon/pull/1548 (which was later reverted)

Reply:
As we are migrating from graphQL to RestAPI, we should rely on RestAPI to get the Suites/checkruns list.

Actually the `checkruns` API returns all check runs including the cirrus ones. So we are good to remove all the cirrus GraphQL logics.

One TODO is distinguish check runs from `cirrus` or `flutter-dashboard`, so that we can have correct task links in the `_FailureDetail`.

Reply:
I created a bug to track this TODO: https://github.com/flutter/flutter/issues/99805.

Reply:
Hi @keyonghan, I discussed with Casey once more on the check runs and found out that it's not necessary to distinguish cirrus to flutter-dashboard.
In the previous logic, the only difference for processing checkRuns and the cirrus is their task links in the _FailureDetail. However, I took a look at the checkRuns script https://api.github.com/repos/flutter/packages/commits/68d068dfda0cf5580290c278bfa3f3b7ff657654/check-runs, and found out that the correct cirrus task link has already been included into the checkRun's details_url. Therefore, we don't need to hardcode its task links as previous logic, we can instead use the checkRun's details_url, the same as line 357 https://github.com/flutter/cocoon/blob/main/app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart#L357

Reply:
That makes it easier. Thanks for the follow up:-)